test/tests/pattern-selection.json

```diff
@@ -0,0 +1,150 @@
+{
+  "$schema": "https://raw.githubusercontent.com/unicode-org/message-format-wg/main/test/schemas/v0/tests.schema.json",
+  "scenario": "Pattern selection",
+  "description": "Tests for the pattern selection parts of the specification",
+  "defaultTestProperties": {
+    "locale": "en-US"
+  },
+  "tests": [
+    {
+      "src": ".match {1 :test:select} 1.0 {{1.0}} 1 {{1}} * {{other}}",
+      "exp": "1"
+    },
+    {
+      "src": ".match {0 :test:select} 1.0 {{1.0}} 1 {{1}} * {{other}}",
+      "exp": "other"
+    },
+    {
+      "src": ".match {$x :test:select} 1.0 {{1.0}} 1 {{1}} * {{other}}",
+      "params": [{ "name": "x", "value": 1 }],
+      "exp": "1"
+    },
+    {
+      "src": ".match {$x :test:select} 1.0 {{1.0}} 1 {{1}} * {{other}}",
+      "params": [{ "name": "x", "value": 2 }],
+      "exp": "other"
+    },
+    {
+      "src": ".input {$x} .match {$x :test:select} 1.0 {{1.0}} 1 {{1}} * {{other}}",
+      "params": [{ "name": "x", "value": 1 }],
+      "exp": "1"
+    },
+    {
+      "src": ".input {$x} .match {$x :test:select} 1.0 {{1.0}} 1 {{1}} * {{other}}",
+      "params": [{ "name": "x", "value": 2 }],
+      "exp": "other"
+    },
+    {
+      "src": ".input {$x :test:select} .match {$x} 1.0 {{1.0}} 1 {{1}} * {{other}}",
+      "params": [{ "name": "x", "value": 1 }],
+      "exp": "1"
+    },
+    {
+      "src": ".input {$x :test:select} .match {$x} 1.0 {{1.0}} 1 {{1}} * {{other}}",
+      "params": [{ "name": "x", "value": 2 }],
+      "exp": "other"
+    },
+    {
+      "src": ".input {$x :test:select} .local $y = {$x} .match {$y} 1.0 {{1.0}} 1 {{1}} * {{other}}",
+      "params": [{ "name": "x", "value": 1 }],
+      "exp": "1"
+    },
+    {
+      "src": ".input {$x :test:select} .local $y = {$x} .match {$y} 1.0 {{1.0}} 1 {{1}} * {{other}}",
+      "params": [{ "name": "x", "value": 2 }],
+      "exp": "other"
+    },
+    {
+      "src": ".match {1 :test:select decimalPlaces=1} 1.0 {{1.0}} 1 {{1}} * {{other}}",
+      "exp": "1.0"
+    },
+    {
+      "src": ".match {1 :test:select decimalPlaces=1} 1 {{1}} 1.0 {{1.0}} * {{other}}",
+      "exp": "1.0"
+    },
+    {
+      "src": ".match {1 :test:select decimalPlaces=9} 1.0 {{1.0}} 1 {{1}} * {{other}}",
+      "exp": "other",
+      "expErrors": [{ "type": "bad-option" }, { "type": "bad-selector" }]
```

Comment:
This one might have better self-documentation. The problem here is that the option _value_ `9` is invalid, a fact that requires knowledge of the `:test:select` function... which is fine. But we could add documentation here such as:

```
* {{bad-option-value}}",
"exp": "bad-option-value",
...
```

Reply:
```suggestion
      "src": ".match {1 :test:select decimalPlaces=9} 1.0 {{1.0}} 1 {{1}} * {{bad-option-value}}",
      "exp": "bad-option-value",
      "expErrors": [{ "type": "bad-option" }, { "type": "bad-selector" }]
```